lib/msf/core/payload/transport_config.rb

```diff
@@ -51,7 +51,7 @@ def transport_config_reverse_http(opts={})
     end
 
     {
-      scheme:      'http',
+      scheme:      datastore['OverrideScheme'] || 'http',
       lhost:       opts[:lhost] || datastore['LHOST'],
```

Comment:
I wonder if adding `lhost:       datastore['OverrideLHOST'] || opts[:lhost] || datastore['LHOST'],` would break everything or fix https://github.com/rapid7/metasploit-framework/issues/6091 ¯\_(ツ)_/¯